### DISCUSSION TITLE
Could not Load dbatools in Linux Container

### DISCUSSION THREAD

**Discussion by aldwinting** [dataplat/dbatools#8643] [Q&A] [answered]
I'm installing dbatools to run on a Ubuntu Container. 

Error: 

     Could not import /usr/local/share/powershell/Modules/dbatools/1.1.142/bin/smo/coreclr/System.Security.SecureString.dll : Exception calling "LoadFrom" with "1" argument(s): "Could not load file or assembly 'System.Security.SecureString, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'."
     At /usr/local/share/powershell/Modules/dbatools/1.1.142/internal/scripts/libraryimport.ps1:150 char:21
     +                     [Reflection.Assembly]::LoadFrom($assemblyPath)
     +                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
     + FullyQualifiedErrorId : FileLoadException
      
     Import-Module : The specified module '/usr/local/share/powershell/Modules/dbatools/1.1.142/bin/dbatools.dll' was not loaded because no valid module file was found in any module directory.
     At /Test5.ps1:12 char:1
     + Import-Module dbatools
     + ~~~~~~~~~~~~~~~~~~~~~~
     + CategoryInfo          : ResourceUnavailable: (/usr/local/share/po\u202642/bin/dbatools.dll:String) [Import-Module], FileNotFoundException
     + FullyQualifiedErrorId : Modules_ModuleNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand

My Dockerfile

  ```
  From myImage
    
    RUN pwsh -Command 'Set-PSRepository -Name PSGallery -InstallationPolicy Trusted'
    
    RUN pwsh -command "Install-Module sqlserver -Scope AllUsers -Force"
    RUN pwsh -command "Install-Module dbatools -Scope AllUsers -Force"
    RUN pwsh -command "Get-module -ListAvailable -Name 'dbatools'"
    RUN pwsh -command "Get-module -ListAvailable -Name 'sqlserver'"
    
    COPY Test5.ps1 .
    
    CMD [ "pwsh", "Test5.ps1"]
```

myImage contains the requred setup to run powershell on ubuntu 20.

Test5.ps1

   ` Import-Module dbatools`

What do I need to do to make my powershell module work without needing to be manually imported?

**THANKS!**

**Comment by potatoqualitee** [comment, marked as answer]
hmm this worked last time i ran it. ill ensure it works with dbatools 2.0

**Reply by potatoqualitee** [reply]
See: https://github.com/dataplat/dbatools/issues/8644

**Reply by aldwinting** [reply]
Thanks!

**Reply by potatoqualitee** [reply]
@aldwinting can you give a whole setup where you encounter this issue? like the docker compose. the FROM my instance, what is the base of that instance?

**Reply by aldwinting** [reply]
Hi, That's actually everything. I haven't started anything yet. Just trying to import it.

**Reply by potatoqualitee** [reply]
oops, i said myisntance, i meant `From myImage` what exactly is your image so that i can try to repro? with my own settings, im able to import dbatools

**Reply by aldwinting** [reply]
It was made by colleagues as our base. It compose of ubuntu:20.04 and PS_VERSION=7.1.0.
